### DISCUSSION TITLE
very very basic guide for setting up kubernetes containers with toolchains?

### DISCUSSION THREAD

**Discussion by cbc02009** [coder/coder#9316] [General]
Is there a step-by-step introductory guide to setting up a dev container with a toolchain and getting started with it? I've found the docs on devcontainers, but I have (unfortunately) little experience with docker so I'm having trouble figuring out what to do from them.

If it helps, I'm running coder from the helm chart, and the workspaces are also using kubernetes. I would like to get a workspace up and running with the rust toolchain (or any toolchain, but I figured this would be an easy one for an example).

 I see that devcontainers has a rust feature here: https://github.com/devcontainers/features/tree/main/src/rust but I'm not sure how to integrate that into the workspace terraform file, or if I'm doing this completely the wrong way and I should be doing something else.

EDIT: actually after reading it again, does the devcontainer get defined in the project, and not the workspace?

**Comment by matifali** [comment]
Coder uses [`envbuilder`](https://github.com/coder/envbuilder) to spin up devcontainers based workspaces and expect you to already have a working `devcontainer` in your repo. To get started on creating a `devcontainer` you can see https://containers.dev. 
And after you have a working devcontainer for your use case you can use the [`devcontainer-kubernetes`](https://github.com/coder/coder/tree/main/examples%2Ftemplates%2Fdevcontainer-kubernetes) starter template. 

If you have any other questions feel free to ask.

**Reply by cbc02009** [reply]
Hello, thank you for your reply. I've read through the documentation on containers.dev, and I'm still confused. I think I'm missing something super obvious. Is there an even more dumbed-down starter guide for devcontainers? Google doesn't really return anything helpful.

Also, once I create the new devcontainer, how do I integrate it into my coder instance using the template?

**Reply by matifali** [reply]
Yes after you have a working Devcontainer you can use the template I linked above.
